### DISCUSSION TITLE
Transfer 06-Sample: How to let provider know data transfer is completed

### DISCUSSION THREAD

**Discussion by busmleon** [eclipse-edc/Samples#49] [Q&A]
Dear EDC-team,
I successfully completed the Http-Pull Sample (Transfer 06) and one question came to my mind:
Is there a way to let the provider know when the transfer was successful so that the transfer will not stay IN_PROGRESS, but rather be also COMPLETED on the provider side (just like on the consumer side)?

Kind regards

**Comment by matinte** [comment]
Hi, I'm also interested on that functionality: I've been looking to class TransferProcessEventListener.java in order to know when a Transfer status changes from STARTED to COMPLETED or TERMINATED. Unfortunately, I dont know how to create instance with following parameters:

    public TransferProcessEventListener(EventRouter eventRouter, Clock clock) {
        this.eventRouter = eventRouter;
        this.clock = clock;
    }

Has anybody a clue about that?

Thanks in advance
